htdocs/takepos/invoice.php

```diff
@@ -786,6 +861,11 @@ function fail($message)
 		$invoice->fetch($placeid);
 	}
 
+	if ($action=="setbatchid") {
+		$sql = "UPDATE ".MAIN_DB_PREFIX."facturedet set special_code=".((int) $batchid)." where rowid=".((int) $idoflineadded);
```

Comment:
The column special_code is already reserved to store information about the nature of line 1=transport, 2=ecotax, ...
So you will have problem if you reuse it to store batch number. So i added 2 new fields into llx_facturedet where you can store:
* field "batch" to store the batch/serial number (a varchar)
* field "fk_warehouse" to store the ID of the warehouse planed to use for stock decrementation

Then line 319, you can replace
`if ($line->special_code > 0) {
$prod_batch->fetch($line->special_code);`
with
`if ($line->batch && $line->fk_warehouse > 0) {
$prod_batch->find(0, '', '', $line->batch, $line->fk_warehouse);`
To get all information of batch stock to use for stock decrease.